### PR TITLE
Better input resizing

### PR DESCRIPTION
**Pull request by klmdb**
fix for [#582](https://github.com/angular-ui/ui-select/issues/582)
- resize input on container resize
- take padding into consideration

### CODE REVIEW

File: src/select.js [hunk 1]
```diff
@@ -513,29 +513,39 @@
       return ctrl.placeholder;
     };
 
-    var containerSizeWatch;
     ctrl.sizeSearchInput = function(){
-      var input = _searchInput[0],
-          container = _searchInput.parent().parent()[0];
+      var input                    = _searchInput[0],
+          container                = _searchInput.parent().parent()[0],
+          matchedElementsContainer = angular.element(input).parent().children()[0];
```
Comment (wesleycho) [inline src/select.js]: This can be rewritten as
```js
matchedElementsContainer = _searchInput.parent().children()[0];
```

File: src/select.js [hunk 2]
```diff
@@ -513,29 +513,39 @@
...
         }
-      }, 0, false);
+
+        var matchedElementsContainerOuterWidth = getSizeProperty(matchedElementsContainer, 'width') || matchedElementsContainer.clientWidth || matchedElementsContainer.offsetWidth;
+
+        return Math.max( 50, containerInnerWidth - matchedElementsContainerOuterWidth - 10);
+      };
+      var setWidth = function(width){
+
+        _searchInput.css('width',width+'px');
```
Comment (wesleycho) [inline src/select.js]: px addition isn't necessary, if it is a number, it is assumed to be in px.

File: src/select.js [hunk 3]
```diff
@@ -513,29 +513,39 @@
...
+
       _searchInput.css('width','10px');
-      var calculate = function(){
-        var newWidth = container.clientWidth - input.offsetLeft - 10;
-        if(newWidth < 50) newWidth = container.clientWidth;
-        _searchInput.css('width',newWidth+'px');
+
+      var getSizeProperty = function(element, property){
+
+        return parseInt(window.getComputedStyle(element, null).getPropertyValue(property) , 10);
```
Comment (wesleycho) [inline src/select.js]: This is a very expensive check - may be a good idea to find a way to cache the information of `window.getComputedStyle` and find a better way to hook into it.

**Comment by Jefiozie** [comment]
Closing it down as this PR is not been updated more than a year

**Pull request closed without merging**